src/init.cpp

```diff
@@ -993,9 +994,10 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
 
     if (!mapMultiArgs["-bip9params"].empty()) {
         // Allow overriding BIP9 parameters for testing
-        if (!Params().MineBlocksOnDemand()) {
-            return InitError("BIP9 parameters may only be overridden on regtest.");
+        if (!Params().AllowsOverriddenSoftFork()) {
```

Comment:
Nit: I think keeping this the way it was is just fine. (Note that `fAllowsOverriddenSoftFork` is not really a chain param, is it?)